src/Cache.php

```diff
@@ -0,0 +1,152 @@
+      $cached_object = self::setCacheObject($product);
+    }
+
+    return $cached_object;
+  }
+
+  /**
+   * Queries cache table for an existing record according to product_id.
+   */
+  private static function queryForProduct($product_id) {
+    global $wpdb;
+    $table_name = $wpdb->prefix . self::PREFIX . '_cache';
+    $sql = $wpdb->prepare("SELECT * FROM {$table_name} WHERE product_id=%d", $product_id);
+    return $wpdb->get_row($sql);
+  }
+
+  /**
+   * Sets this product entry in the cache table.
+   */
+  private static function setCacheObject($product) {
+    // Gets all attachments for this product slider.
+    $images = self::getProductSliderImageIds($product);
+
+    $slider_image_size = has_image_size('woocommerce_thumbnail') ? 'woocommerce_thumbnail' : 'medium';
+    $slider_image_src = apply_filters('gallerya/image_size_product_variation_slider', $slider_image_size);
+    $transparent_pixel = 'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7';
+
+    $queried_data = [];
+    foreach ($images as $index => $image) {
+      $queried_data[$index]['image_markup'] = wp_get_attachment_image($image->ID, $slider_image_src, FALSE, $index ? [
+        'src' => $transparent_pixel,
+        'srcset' => $transparent_pixel,
+        'data-flickity-lazyload-src' => wp_get_attachment_image_url($image->ID, $slider_image_src),
+        'data-flickity-lazyload-srcset' => wp_get_attachment_image_srcset($image->ID, $slider_image_src),
+      ] : []);
+    }
+
+    // Injects cached object.
+    self::injectQueriedData($queried_data, $product->get_id());
+
+    // Returns queried data.
+    return $queried_data;
+  }
+
+  /**
+   * Returns all attachments for this product slider.
+   */
+  private static function getProductSliderImageIds($product) {
+    // Stores ttachments ids.
+    $attachment_ids = [];
+
+    // Sets the main product image as first entry.
+    $attachment_ids[] = $product->get_image_id();
+
+    // Gets the first image of each product variation.
+    $variations = $product->get_available_variations();
+    foreach ($variations as $variation) {
+      $attachment_ids[] = $variation['image_id'];
+    }
+    $attachment_ids = array_unique($attachment_ids);
+
+    // Query for the post type attachments.
+    $args['post_type'] = 'attachment';
+    $args['include'] = $attachment_ids;
+    $args['orderby'] = 'post__in';
+
+    // Returns attachments collection.
+    return get_posts($args);
+  }
+
+  /**
+   * Inserts or updates product record in cache table.
+   */
+  private static function injectQueriedData($queried_data, $product_id) {
+    global $wpdb;
+    $table_name = $wpdb->prefix . self::PREFIX . '_cache';
+
+    // Updates existing record if found.
+    $existing_record = self::queryForProduct($product_id);
+    if (!empty($existing_record)) {
+      $sql = $wpdb->prepare("UPDATE {$table_name} SET query_data=%s WHERE product_id=%d", json_encode($queried_data), $product_id);
```

Comment:
Key missmatch? `query_data` vs `queried_data` (https://github.com/netzstrategen/wordpress-gallerya/pull/44/files#diff-010266d5c06d24ba1968dd13647dc805R148)

Reply:
IMO `data` or `value` as key name is sufficient.